Assignment2-Python/stackmin.py

```diff
@@ -0,0 +1,39 @@
+#3.2 Stack Min (p.98)
+'''Stack Min: How would you design a stack which, in addition to p u s h and pop, has a function m i n
+which returns the minimum eiement? Push, p o p and m i n should ail operate in 0 ( 1 ) time'''
+
+class MinStack:
+    def __init__(self) -> None:
+        self.stack = []
+        self.min_stack = []
+    
+    def push(self, val: int) -> None:
+
+        self.stack.append(val)
+        if not self.min_stack or val <= self.min_stack[-1]:
+            self.min_stack.append(val)
+        else: 
+            self.min_stack.append(self.min_stack[-1])
```

Comment:
Because lines 14 and 16 are almost identical, you can do this in one line using a ternary. Also note that `min` is a thing in Python, and might be a cleaner way to implement something like this.
```suggestion
        self.min_stack.append(self.min_stack[-1] if self.min_stack and val <= self.min_stack[-1] else val)
```